src/CalcViewModel/Snapshots.h

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <vector>
+#include <memory>
+
+#include "CalcManager/CalculatorManager.h"
+
+namespace CalculatorApp::ViewModel::Snapshot
+{
+public
+    interface struct ICalcManagerIExprCommand
+    {
+    };
+
+public
+    ref struct UnaryCommand sealed : public ICalcManagerIExprCommand
+    {
+        property Windows::Foundation::Collections::IVectorView<int> ^ Commands {
+            Windows::Foundation::Collections::IVectorView<int> ^ get();
+            void set(Windows::Foundation::Collections::IVectorView<int> ^ commands);
+        };
+
+        UnaryCommand();
+
+        internal :;
+        explicit UnaryCommand(std::vector<int> cmds);
+        std::vector<int> m_cmds;
+    };
+
+public
+    ref struct BinaryCommand sealed : public ICalcManagerIExprCommand
+    {
+        property int Command;
+
+        BinaryCommand();
+
+        internal :;
+        explicit BinaryCommand(int cmd);
+    };
+
+public
+    ref struct OperandCommand sealed : public ICalcManagerIExprCommand
+    {
+        property bool IsNegative;
+        property bool IsDecimalPresent;
+        property bool IsSciFmt;
+        property Windows::Foundation::Collections::IVectorView<int> ^ Commands {
+            Windows::Foundation::Collections::IVectorView<int> ^ get();
+            void set(Windows::Foundation::Collections::IVectorView<int> ^ commands);
+        };
+
+        OperandCommand();
+
+        internal :;
+        explicit OperandCommand(bool isNegative, bool isDecimal, bool isSciFmt, std::vector<int> cmds);
+        std::vector<int> m_cmds;
+    };
+
+public
+    ref struct Parentheses sealed : public ICalcManagerIExprCommand
+    {
+        property int Command;
+
+        Parentheses();
+
+        internal :;
+        explicit Parentheses(int cmd);
+    };
+
+public
+    ref struct CalcManagerHistoryToken sealed
```

Comment:
The name is kind of confusing as this token is not only used in history but also in `ExpressionDisplaySnapshot`. Perhaps we can just name it `CalcManagerToken` without history?

Reply:
make sense. updated